File: slidingwindow.c
#include <stdio.h>
/**
 * Calculate the length of the longest subarray
 * arr: the array we have to find subarray for
 * Return: Length of subarray
 */
int find_length(int *arr, int k, int size)
{
	int left = 0;
	int right = 0;
	int i = 0;
	int sums = 0;
	int length = 0;
	for(right = 0; right < size; right++)
	{
		sums += arr[right];
		while (sums > k)
		{
			sums -= arr[left];
			left ++;
		}
		i = right - left + 1;
		if (length < i)
		{
			length = i;
		}
	}
	return length;
}
int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int k = 11;
    int size = sizeof(arr) / sizeof(arr[0]);

    int result = find_length(arr, k, size);

    printf("Length of the longest subarray with sum less than or equal to %d: %d\n", k, result);

    return 0;
}

File: sliding_window.py
#!/usr/bin/python3
"""Use the concept of sliding window to find the maximum subarray"""
from typing import List


def find_length(nums: List[int], k: int) -> int:
    left = sums = length = 0
    for right in range (len(nums)):
        sums += nums[right]
        while sums > k:
            sums -= nums[left]
            left += 1
        length = max(length, right - left + 1)
    return length


nums2 = [2, -1, 2, 1, -3, 2, 1]ls

k2 = 4
result2 = find_length(nums2, k2)
print(f"Test 2: {result2}")  # Expected output: 5